clients/nodejs/validator-registration.js

```diff
@@ -0,0 +1,338 @@
+    const recipientAddr = Nimiq.Address.fromUserFriendlyAddress('NQ07 0000 0000 0000 0000 0000 0000 0000 0000');
+
+    // Extract the validator configuration (we read the private keys from the config file)
+    let validatorPrivateKey = Nimiq.PrivateKey.unserialize(Nimiq.BufferUtils.fromHex(validator_config.ValidatorAccount.PrivateKey));
+    let signingPrivateKey = Nimiq.PrivateKey.unserialize(Nimiq.BufferUtils.fromHex(validator_config.SigningKey.PrivateKey));
+    let votingSecretKey =  NimiqPOS.BLSSecretKey.fromHex(validator_config.VotingKey.SecretKey);
+
+    // Create the KeyPairs
+    const validatorKeyPair = Nimiq.KeyPair.derive(validatorPrivateKey);
+    const SigningKeyPair = Nimiq.KeyPair.derive(signingPrivateKey);
+    const votingKeyPair = NimiqPOS.BLSKeyPair.derive(votingSecretKey);
+
+    // Create a new validator registry
+    let validator = new ValidatorRegistry(validatorKeyPair.publicKey.toAddress(), SigningKeyPair.publicKey, votingKeyPair.publicKey);
+
+    // Get the transaction's data
+    let data = validator.get_serialized_data();
+
+    // We monitor the status of the validator registration transactions
+    $.client.addTransactionListener((tx) => {
+        console.log(' Transaction update: ');
+        console.log(tx);
+    }, [recipientAddr]);
+
+    let consensusState = Nimiq.Client.ConsensusState.CONNECTING;
+    $.client.addConsensusChangedListener(async (state) => {
+        consensusState = state;
+        if (state === Nimiq.Client.ConsensusState.ESTABLISHED) {
+            Nimiq.Log.i(TAG, `Blockchain ${config.type}-consensus established in ${(Date.now() - START) / 1000}s.`);
+            const chainHeight = await $.client.getHeadHeight();
+            const chainHeadHash = await $.client.getHeadHash();
+            Nimiq.Log.i(TAG, `Current state: height=${chainHeight}, headHash=${chainHeadHash}`);
+
+            // Once we obtain consensus, we send the validator transactions
+            // Send the txns for validator registration
+            for (let i = 0; i < 6; i++) {
+                console.log('sending transaction: ');
+
+                let transaction = new Nimiq.ExtendedTransaction(validatorKeyPair.publicKey.toAddress(), Nimiq.Account.Type.BASIC, recipientAddr, Nimiq.Account.Type.BASIC, 10, 10, chainHeight, Nimiq.Transaction.Flag.NONE, data[i]);
+                let proof = new Nimiq.SerialBuffer(Nimiq.SignatureProof.SINGLE_SIG_SIZE);
+                Nimiq.SignatureProof.singleSig(validatorKeyPair.publicKey, Nimiq.Signature.create(validatorKeyPair.privateKey, validatorKeyPair.publicKey, transaction.serializeContent())).serialize(proof);
+                transaction.proof = proof;
+
+                let result = await $.client.sendTransaction(transaction);
+
+                console.log(' Transaction result: ');
+                console.log(result);
+
+                console.log(' Transaction hash: ');
+                console.log(transaction.hash().toHex());
```

Comment:
Each further transaction sending should maybe fail, if a tx fails to send? E.g. if it reports as `result.state === 'new'`? 

If the txs are all in the same block, their order is not guaranteed I think. Is the PoS genesis-generating script able to work with unsorted transactions?

Reply:
Good question, since they are 0 value txns they should not fail..
 The PoS generation script is able to work with unsorted transactions:
 As part of the encoding that we created for the tx data, we are using 1 byte as "type" which is just a number to indicate the order of the registration txn. So the migration tool will look this number to sort the registration txns.